packages/pigeon_build_core/lib/src/implementations/build_handler.dart

```diff
@@ -220,10 +225,10 @@ class PigeonBuildHandler {
     } else if (isBase || basePath == null) {
       result = path;
     } else {
-      result = p.join(basePath, path);
+      result = pathContext.join(basePath, path);
```

Comment:
I don't think anything necessarily needs to change there. If the path starts with `/` it will continue to treat it as relative to the project root, just as before.

I'm not quite sure what you mean by `combinePath` running independently?